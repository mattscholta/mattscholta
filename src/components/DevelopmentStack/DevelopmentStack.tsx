import * as React from 'react';
import classnames from 'classnames';

import styles from './styles.module.css';

export interface DevelopmentStackProps {
  className?: string;
}

/**
 * @name DevelopmentStack
 * @description Stateless Component / Pure Function
 */
const DevelopmentStack: React.FC<DevelopmentStackProps> = (props) => {
  const { className } = props;

  // Styles
  const css = classnames(styles.component, className);

  return (
    <div className={css}>
      <h2 className="ui-heading medium">Technology / Tools</h2>
      <ul className="ui-list-tags">
        <li>Apollo</li>
        <li>Adobe Creative Suite</li>
        <li>Agile</li>
        <li>AWS</li>
        <li>Babel</li>
        <li>CSS</li>
        <li>Docker</li>
        <li>ElasticSearch</li>
        <li>Express</li>
        <li>Figma</li>
        <li>Google Analytics</li>
        <li>Google Tag Manger</li>
        <li>Git / Github</li>
        <li>GraphQL</li>
        <li>HTML</li>
        <li>Javascript</li>
        <li>JWT</li>
        <li>Kubernetes</li>
        <li>Lerna</li>
        <li>Node.js</li>
        <li>NoSQL</li>
        <li>NPM</li>
        <li>PostCSS</li>
        <li>React</li>
        <li>RecoilJS</li>
        <li>Redux</li>
        <li>REST</li>
        <li>Rollup</li>
        <li>SQL / MySQL</li>
        <li>Sketch</li>
        <li>Swarm</li>
        <li>TypeScript</li>
        <li>Webpack</li>
      </ul>
    </div>
  );
};

export { DevelopmentStack };
