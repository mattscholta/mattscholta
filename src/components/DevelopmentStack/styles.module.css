.component,
.component h2 {
  margin-bottom: var(--grid-4x);
}

.component :global(.ui-list-tags) li {
  cursor: pointer;
  transition: color 240ms ease-out;
}
