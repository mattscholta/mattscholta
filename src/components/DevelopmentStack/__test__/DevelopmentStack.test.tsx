import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { DevelopmentStack, DevelopmentStackProps } from '../DevelopmentStack';

describe('DevelopmentStack', () => {
  let props: DevelopmentStackProps;
  let wrapper: RenderResult;

  const createWrapper = () => <DevelopmentStack {...props} />;

  beforeEach(() => {
    props = {};
    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
