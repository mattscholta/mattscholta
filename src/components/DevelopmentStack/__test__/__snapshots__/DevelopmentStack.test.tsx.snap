// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DevelopmentStack matches snapshot 1`] = `
NodeList [
  <div
    class="component"
  >
    <h2
      class="ui-heading medium"
    >
      Technology / Tools
    </h2>
    <ul
      class="ui-list-tags"
    >
      <li>
        Apollo
      </li>
      <li>
        Adobe Creative Suite
      </li>
      <li>
        Agile
      </li>
      <li>
        AWS
      </li>
      <li>
        Babel
      </li>
      <li>
        CSS
      </li>
      <li>
        Docker
      </li>
      <li>
        ElasticSearch
      </li>
      <li>
        Express
      </li>
      <li>
        Figma
      </li>
      <li>
        Google Analytics
      </li>
      <li>
        Google Cloud
      </li>
      <li>
        Google Tag Manger
      </li>
      <li>
        Git / Github
      </li>
      <li>
        GraphQL
      </li>
      <li>
        HTML
      </li>
      <li>
        Javascript
      </li>
      <li>
        JWT
      </li>
      <li>
        Kubernetes
      </li>
      <li>
        Lerna
      </li>
      <li>
        NextJS
      </li>
      <li>
        Node.js
      </li>
      <li>
        NoSQL
      </li>
      <li>
        NPM
      </li>
      <li>
        PostCSS
      </li>
      <li>
        Postgres
      </li>
      <li>
        React
      </li>
      <li>
        Remix ðŸ¤©
      </li>
      <li>
        RecoilJS
      </li>
      <li>
        Redux
      </li>
      <li>
        REST
      </li>
      <li>
        Rollup
      </li>
      <li>
        SQL / MySQL
      </li>
      <li>
        Sketch
      </li>
      <li>
        Swarm
      </li>
      <li>
        TypeScript
      </li>
      <li>
        Webpack
      </li>
    </ul>
  </div>,
]
`;
